aya/src/obj/mod.rs

```diff
@@ -374,6 +409,75 @@ impl Object {
 
         Ok(())
     }
+
+    fn parse_map(
+        &self,
+        section: &Section,
+        name: &str,
+        globals: &HashMap<&str, &[u8]>,
+    ) -> Result<Map, ParseError> {
+        let (def, data, kind) =
+            if name == ".bss" || name.starts_with(".data") || name.starts_with(".rodata") {
+                let mut data = section.data.to_vec();
+                let symbols: Vec<&Symbol> = self
+                    .symbols_by_index
+                    .iter()
+                    .filter(|(_, s)| {
+                        if let Some(idx) = s.section_index {
+                            idx == section.index
+                        } else {
+                            false
+                        }
+                    })
+                    .map(|(_, s)| s)
+                    .collect();
+                for symbol in symbols {
```

Comment:
You should do the opposite, iterate on globals as the number of globals is always going to be stricly smaller than the number of symbols

Reply:
And also if a global can't be found in the symbol table, you should error out